src/RelativeTime/RelativeTime.tsx

```diff
@@ -0,0 +1,11 @@
+import React from 'react'
+import styled from 'styled-components'
+import {RelativeTimeElement} from '@github/time-elements'
```

Comment:
given this import now calls 

window.customElements.define in module scope, what type of story are we providing for users who use this in react, but also use server side rendering?

Reply:
I think we can stub out `HTMLElement` and `window.customElement.[get/has/define]` within an SSR context fairly simply, which should make this a painless lift. I don't foresee any major consequences to that, apart from perhaps pushback from folks wanting that in the SSR runtime codebase. If there is sufficient pushback then we can resort to having a browser only bundle in Primer React, but I think that should be a last resort.

Reply:
For non-github consumers, that stub pattern will likely be un-ideal (and even for us - I'd be hesitant to stubbing those). 

a client bundle makes sense, although I don't love that complexity add either (especially given the tradeoffs on how that bundle codesplits (one big import with things people may never use, where lazy loading may not be ideal either for those wanting single bundles vs many small imports needing to be managed separately)

but glad we have some ideas in the works for it

Reply:
Running multi-modal JS (or "isomorphic JS" or "Universal JS") faces these problems writ large, and is something we'll need to grapple with in the broader context, not just for those two globals but others too. An engineer can trivially reach for globals like `ResizeObserver`, `EventTarget`, `performance.now`, `URL`, `fetch`, `console.log`, `crypto.subtle`. While all work in a browser most won't in NodeJS. Deno, however, provides all except `ResizeObserver`. I think this illustrates an interesting juncture where some philosophies are fixed towards making application code renderer-agnostic, but other philosophies are looking towards making run-times more similar to browsers. Node is gradually leaning more towards providing these globals too - it recently added `EventTarget`.

On the other hand, yes, client bundles make more and more sense if we consider that not every element on the page needs a client side component definition. As we start to move away from that idea within React, a natural conclusion may be what other frameworks like Rails have done: render components server side, and deliver a handful of interactive ones as dedicated client components. This moves away from multi-modal JS ("Isomorphic JS"/"Universal JS") as it demarcates that some components are designed for consumption within the server and some are not.

Both choices take us down one path, and come with varying trade-offs.

Reply:
I think React is definitely trying to have dedicated client and server components based on the RFC and Next.js work, but I think that client components will still require the ability to "pre-render" on the server.

This seems like something we should aim to support in order to make sure a component renders something usable when delivered to the client and can be enhanced as JavaScript loads and the component is hydrated. One thing I would like to avoid are components that noop on the server and then have to wait for hydration on the client in order to render something onto the page.

Curious what you all think about this 🤔

In the interim, for things like `HTMLElement` would a shim like what they do in lit feel appropriate? ([Link](https://github.com/lit/lit/blob/a6a5591327a90e9c7d922d1d978627794909c422/packages/reactive-element/src/reactive-element.ts#L395-L398)) Then for things like `customElements.define` would it be feasible to check to see if the runtime supports it before calling it?